### ISSUE TITLE
Example: how to implement template variable support in data source

### ISSUE THREAD

**Issue by mckn** [grafana/grafana-plugin-examples#141]
The available method on the datasource plugin class is marked as deprecated. Without any examples or documentation on alternative approaches. We should provide some examples covering the different ways of adding a template variable support to data sources.

This is how you add the variable support to the data source:
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana-testdata-datasource/datasource.ts#L37

This is an example of how you implement the standard variable support:
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana-testdata-datasource/variables.ts

StandardVariableSupport = Extend this class in a data source plugin to use the standard query editor for Query variables
CustomVariableSupport = Extend this class in a data source plugin to use a customized query editor for Query variables
DataSourceVariableSupport = Extend this class in a data source plugin to use the query editor in the data source plugin for Query variables

**Comment by mckn** [comment]
We already have this one in the grafana/grafana repository https://github.com/grafana/grafana/issues/63619

**Comment by shelldandy** [comment]
Do we have an example coming from a class with `DataSourceWithBackend` ?

**Comment by mckn** [comment]
No, I don't think we have any example. We should add one to this [repository](https://github.com/grafana/grafana-plugin-examples) before closing this issue.

**Comment by academo** [comment]
We are no longer writing examples to demonstrate code and instead prefer to write documentation about the topics. 

Since this repository example's are now auto generated from create-plugin I am closing this issues.
